src/Orleans/AssemblyLoader/AssemblyLoader.cs

```diff
@@ -82,27 +82,26 @@ internal class AssemblyLoader
             return discoveredAssemblyLocations;
         }
 #endif
-        public static T TryLoadAndCreateInstance<T>(string assemblyName, Logger logger) where T : class
+        public static T TryLoadAndCreateInstance<T>(string assemblyName, Logger logger, IServiceProvider serviceProvider) where T : class
         {
             try
             {
                 var assembly = Assembly.Load(new AssemblyName(assemblyName));
                 var foundType =
                     TypeUtils.GetTypes(
                         assembly,
-                        type =>
-                        typeof(T).IsAssignableFrom(type) && !type.GetTypeInfo().IsInterface
-                        && type.GetConstructor(Type.EmptyTypes) != null, logger).FirstOrDefault();
+                        type => typeof(T).IsAssignableFrom(type) && !type.GetTypeInfo().IsInterface,
+                        logger).FirstOrDefault();
                 if (foundType == null)
                 {
                     return null;
                 }
 
-                return (T)Activator.CreateInstance(foundType, true);
+                return (T)ActivatorUtilities.CreateInstance(serviceProvider, foundType);
             }
-            catch (FileNotFoundException exception)
+            catch (FileNotFoundException)
             {
-                logger.Warn(ErrorCode.Loader_TryLoadAndCreateInstance_Failure, exception.Message, exception);
+                logger.Info(ErrorCode.Loader_TryLoadAndCreateInstance_Failure, $"Failed to find assembly {assemblyName} to create instance of {typeof(T)}.");
```

Comment:
This should never have been a warning. The method is explicitly called TryLoad..., and it looks very ugly in user's logs (it spews red text on startup). Hopefully it's fine to fix this here instead of creating a new PR.